java-frontend/src/main/java/org/sonar/java/resolve/SecondPass.java

```diff
@@ -109,6 +109,12 @@ private void complete(JavaSymbol.TypeJavaSymbol symbol) {
 
     if ((symbol.flags() & Flags.INTERFACE) == 0) {
       symbol.members.enter(new JavaSymbol.VariableJavaSymbol(Flags.FINAL, "super", type.supertype, symbol));
+    } else {
+      // JLS9 - 15.12.1 : To support invocation of default methods in superinterfaces, the TypeName may also refer
```

Comment:
Wording is not really correct, and a bit ambiguous. `TypeName` is not defined. I would go for

> Used in form 'TypeName.super.foo()', where 'TypeName' is an interface. To support invocation of default methods **from** super-interfaces, 'TypeName' may also refer to a direct super-interface of the current class or interface. **The method being invoked ('foo()') has to be searched in that super-interface**.